test/unit/org/apache/cassandra/cql3/statements/DescribeStatementTest.java

```diff
@@ -377,6 +393,52 @@ public void testDescribe() throws Throwable
         }
     }
 
+    private static volatile CountDownLatch describeStatementLatch;
+    public static void awaitDescribeStatementLatch()
+    {
+        if (Arrays.stream(Thread.currentThread().getStackTrace()).anyMatch(ste -> ste.getClassName().equals(DescribeStatement.class.getName())))
+        {
+            describeStatementLatch.decrement();
+            describeStatementLatch.awaitThrowUncheckedOnInterrupt();
+        }
+    }
+
+    @Test
+    @BMRule(name = "blablabla",
+            targetClass = "Schema",
+            targetMethod = "distributedAndLocalKeyspaces",
+            targetLocation = "AT ENTRY",
+            action = "org.apache.cassandra.cql3.statements.DescribeStatementTest.awaitDescribeStatementLatch()")
+    public void testNotReturningPartiallyAppliedSchema() throws Throwable
+    {
+        describeStatementLatch = CountDownLatch.newCountDownLatch(2);
+        ForkJoinTask<ResultSet> describeFuture = ForkJoinPool.commonPool().submit(() -> {
+            try
+            {
+                return executeNetWithPaging(ProtocolVersion.CURRENT, "DESCRIBE KEYSPACES", 1);
+            }
+            catch (Throwable e)
+            {
+                throw new RuntimeException(e);
+            }
+        });
+        // wait for the describe statement to be in progress
+        Awaitility.await().atMost(Duration.ofSeconds(5)).until(() -> describeStatementLatch.count() == 1);
```

Comment:
Schema cahnge operations can take >60s in jenkins. We have 120s in dtests i.e. and I am raising this to 120s for jvm dtests. We should raise it here as well to prevent flakies.

Reply:
Though, this is not a dtest and there is no networking. Repeated run with this test already completed without problems. If you insist, I can change that, however, I cannot justify such a long timeout given what it waits for.

Reply:
_jvm_ dtest, not py dtests. And repeated runs worked on circle, not jenkins. I know it sounds crazy but on jenkins schema changes can take >60s easily and sbdy will have to read logs/debug if we don't go with more realistic, though crazy, timeouts :shrug:

Reply:
this is cqltester based unit test....

Reply:
Yes I know.

Reply:
changed to 20 seconds, it does not make sense to increase it more for those tests. Also renamed "blablabla" rule 🤦🏻

Reply:
It does indeed make a lot of sense to raise it. The pain of having to continuously debug flakies, read the logs, chase the code, etc. which in the end happen to be just timeouts it's not negligible and very demotivating imo.

20s sounds like a magic number. You mentioned `test.driver.read_timeout_ms` is at 24s. I wonder if we can access that programmatically?